lhotse/supervision.py

```diff
@@ -97,13 +175,51 @@ def transform_text(self, transform_fn: Callable[[str], str]) -> 'SupervisionSegm
         if self.text is None:
             return self
         return fastcopy(self, text=transform_fn(self.text))
+    
+    def transform_alignment(
+        self,
+        transform_fn: Callable[[str], str],
+        type: Optional[str] = 'word'
+    ) -> 'SupervisionSegment':
+        """
+        Return a copy of the current segment with transformed ``alignment`` field.
+        Useful for text normalization, phonetic transcription, etc.
+
+        :param type:  alignment type to transform (key for alignment dict).
+        :param transform_fn: a function that accepts a string and returns a string.
+        :return: a ``SupervisionSegment`` with adjusted alignments.
+        """
+        if self.alignment is None:
+            return self
+        return fastcopy(
+            self,
+            alignment={
+                ali_type:[
+                    item.transform(transform_fn=transform_fn)
+                    if ali_type == type else item
+                    for item in ali
+                ]
+                for ali_type, ali in self.alignment.items()
+            } if self.alignment else None
+        )
 
     def to_dict(self) -> dict:
-        return asdict_nonull(self)
+        """
+        We use the recursive conversion only if alignments are present, since it may
+        potentially be slower due to type checking of member objects.
+        """
+        return asdict_nonull(self) if self.alignment is None else asdict_nonull_recursive(self)
```

Comment:
The `AlignmentItem` class is no longer a subclass of `NamedTuple`, is `asdict_nonull_recursive` really necessary? It makes the code slow, I believe.

Also, the code in `cut.py` should also be updated to remove `asdict_nonull_recursive`.

Reply:
Good point, @desh2608 could you address Fangjuns comments?

Btw @csukuangfj we should make sure that Lhotse alignments and Snowfall alignments interact well together. If there are any additions/changes that are helpful on Lhotse side, please let us know.